File: spring-reactive/src/main/java/dev/nonava/techstacks/spring/reactive/greeting/application/service/GreetingService.java
/*
 * Copyright 2021 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */

package dev.nonava.techstacks.spring.reactive.greeting.application.service;

import dev.nonava.techstacks.spring.reactive.greeting.domain.model.Greeting;
import dev.nonava.techstacks.spring.reactive.greeting.domain.model.Username;
import org.springframework.stereotype.Service;
import reactor.core.publisher.Flux;

@Service
public class GreetingService {
    public Flux<Greeting> listGreetings() {
        var greeting = new Greeting(new Username("World"));
        return Flux.just(greeting);
    }
}

File: nest/src/greeting/application/service/greeting.service.ts
/*
 * Copyright 2021-2022 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */

import {Injectable} from "@nestjs/common";
import {Greeting} from "../../domain/model/greeting";
import {Result} from "../../../core/domain/model/result";
import {Username} from "../../domain/model/username";

@Injectable()
export class GreetingService {
    listGreetings(): Result<Greeting[], Error> {
        let greeting = new Greeting(new Username('World'));
        return Result.of([greeting]);
    }
}
